.github/workflows/platform-docker-build-test-publish.yml

```diff
@@ -0,0 +1,161 @@
+      target: oss-unified
+      image-name: flagsmith
+
+  docker-build-private-cloud-api:
+    name: Build Private Cloud API Image
+    uses: ./.github/workflows/.reusable-docker-build.yml
+    with:
+      target: private-cloud-api
+      image-name: flagsmith-private-cloud-api
+    secrets:
+      secrets: |
+        github_private_cloud_token=${{ secrets.GH_PRIVATE_ACCESS_TOKEN }}
+
+  docker-build-private-cloud:
+    name: Build Private Cloud Image
+    uses: ./.github/workflows/.reusable-docker-build.yml
+    with:
+      target: private-cloud-unified
+      image-name: flagsmith-private-cloud
+    secrets:
+      secrets: |
+        github_private_cloud_token=${{ secrets.GH_PRIVATE_ACCESS_TOKEN }}
+
+  docker-build-e2e:
+    name: Build E2E Image
+    uses: ./.github/workflows/.reusable-docker-build.yml
+    with:
+      file: frontend/Dockerfile.e2e
+      image-name: flagsmith-e2e
+      scan: false
+
+  run-e2e-tests:
+    needs: [docker-build-api, docker-build-private-cloud-api, docker-build-e2e]
+    uses: ./.github/workflows/.reusable-docker-e2e-tests.yml
+    with:
+      api-image: ${{ matrix.api-image }}
+      e2e-image: ${{ needs.docker-build-e2e.outputs.image }}
+
+    strategy:
+      matrix:
+        api-image:
+          - ${{ needs.docker-build-api.outputs.image }}
+          - ${{ needs.docker-build-private-cloud-api.outputs.image }}
+
+  docker-publish-api:
+    needs: [docker-build-api, run-e2e-tests]
+    uses: ./.github/workflows/.reusable-docker-publish.yml
+    if: github.event_name == 'release'
+    with:
+      source-images: ${{ needs.docker-build-api.outputs.image }}
+      target-images: flagsmith/flagsmith-api
+
+  docker-publish-frontend:
+    needs: [docker-build-frontend, run-e2e-tests]
+    uses: ./.github/workflows/.reusable-docker-publish.yml
+    if: github.event_name == 'release'
+    with:
+      source-images: ${{ needs.docker-build-frontend.outputs.image }}
+      target-images: flagsmith/flagsmith-frontend
+
+  docker-publish-unified:
+    needs: [docker-build-unified, run-e2e-tests]
+    uses: ./.github/workflows/.reusable-docker-publish.yml
+    if: github.event_name == 'release'
+    with:
+      source-images: ${{ needs.docker-build-unified.outputs.image }}
+      target-images: flagsmith/flagsmith
+
+  docker-publish-private-cloud-api:
+    needs: [docker-build-private-cloud-api, run-e2e-tests]
+    uses: ./.github/workflows/.reusable-docker-publish.yml
+    if: github.event_name == 'release'
+    with:
+      source-images: ${{ needs.docker-build-private-cloud-api.outputs.image }}
+      target-images: flagsmith/flagsmith-private-cloud-api
+
+  docker-publish-private-cloud:
+    needs: [docker-build-private-cloud, run-e2e-tests]
+    uses: ./.github/workflows/.reusable-docker-publish.yml
+    if: github.event_name == 'release'
+    with:
+      source-images: ${{ needs.docker-build-private-cloud.outputs.image }}
+      target-images: flagsmith/flagsmith-private-cloud
+
+  update-charts:
```

Comment:
If possible, I'd prefer to see this in a separate workflow, but I can't remember if you can pass variables / add dependencies between workflows?

Reply:
You can, but there's so much boilerplate required to support it I'm hesitant to do it.

I prefer to have all interdependent jobs in one workflow anyway.

Reply:
Yep, sure 👍